### PR TITLE
Flood Visualization

### LINKED ISSUE #7: Flood Detection Visualization
**Is your feature request related to a problem? Please describe.**

Add something to the GUI to show pilots flooding has occurred.

Done in parallel with #6.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Tazmania40**
# Pull Request

## What type of PR is this? (check all applicable)

- [X] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adds functionality that detects flooding on a ROS2 topic /flooding and displays that information on the pilot display as a red dot rather than the default green state. The functionality is latching until the GUI is reset, as afaik, there isn't really a reason where a flood would not cause a system reset of some kind... 

## Related Tickets & Documents

- Closes #7

### CODE REVIEW

File: src/surface/gui/gui/widgets/flood_visual.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+class FloodVisual(QWidget):
+
+    signal: pyqtSignal = pyqtSignal(Flooding)
+
+    def __init__(self):
+        super().__init__()
+        # Boilerplate PyQt Setup to link to ROS through a signal/subscriber
+        self.signal.connect(self.refresh)
+        self.subscription: GUIEventSubscriber = GUIEventSubscriber(Flooding,
```
Comment (InvincibleRMC) [inline src/surface/gui/gui/widgets/flood_visual.py:17]: No forward /

**Review by InvincibleRMC — CHANGES_REQUESTED** [review]
Add remapping for flooding topic and update README.md for gui package.

**Comment by InvincibleRMC** [comment]
On linux the Qlabel of emojis does not seem to work. Surely there is some QShape thing that can be used as circles or something.

**Comment by Tazmania40** [comment]
Appears emoji support is more spotty than I thought... There unfortunately isn't a good way to do a QShape as described, but I'll look into it further.

**Comment by InvincibleRMC** [comment]
> Appears emoji support is more spotty than I thought... There unfortunately isn't a good way to do a QShape as described, but I'll look into it further.

I think something from here might work.

https://likegeeks.com/pyqt5-drawing-tutorial/#Draw_a_circle

File: src/surface/gui/gui/widgets/flood_visual.py (hunk 1 above)
Comment (Tazmania40) [reply src/surface/gui/gui/widgets/flood_visual.py:17]: Fixed

File: src/surface/gui/gui/widgets/flood_visual.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        self.mylayout.addWidget(self.indicator)
+        self.setLayout(self.mylayout)
+
+    @pyqtSlot(Flooding)
+    def refresh(self, msg: Flooding) -> None:
+        if Flooding.flooding:
+            self.indicator.setText('FLOODING')
+            font: QFont = QFont("Arial", 14)
+            self.indicator.setFont(font)
+            self.subscription.get_logger().error("Robot is actively flooding, do something!")
```
Comment (benjaminwp18) [inline src/surface/gui/gui/widgets/flood_visual.py:41]: Add an `else` clause that removes the flooding warning if there's no flooding
Remove the font changing code here; it's now unused

File: src/surface/gui/gui/widgets/flood_visual.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                                                                   'flooding',
+                                                                   self.signal)
+        # Create basic 2 vertical stacked boxes layout
+        self.mylayout = QVBoxLayout()
+        # Create the label that tells us what this is
+        self.label: QLabel = QLabel('Flooding Indicator')
+        # Set font and size
+        font: QFont = QFont("Arial", 14)
+        self.label.setFont(font)
+        self.mylayout.addWidget(self.label)
```
Comment (benjaminwp18) [inline src/surface/gui/gui/widgets/flood_visual.py:27]: Remove redundant comments
Rename `self.mylayout` to something else. `self.flood_layout`?

File: src/surface/gui/gui/widgets/flood_visual.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# flooding (boolean) (True is flooding)
+from rov_msgs.msg import Flooding
+from PyQt6.QtWidgets import QWidget, QVBoxLayout, QLabel
+from PyQt6.QtGui import QFont
+from gui.event_nodes.subscriber import GUIEventSubscriber
+from PyQt6.QtCore import pyqtSignal, pyqtSlot
+
+
+class FloodVisual(QWidget):
```
Comment (benjaminwp18) [inline src/surface/gui/gui/widgets/flood_visual.py:9]: `FloodVisual` sounds awkward to me. `FloodWarning`? `FloodAlertWidget`?

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

File: src/surface/gui/gui/widgets/flood_visual.py (hunk 3 above)
Comment (Tazmania40) [reply src/surface/gui/gui/widgets/flood_visual.py]: renamed, "redundant comments removed"

File: src/surface/gui/gui/widgets/flood_visual.py (hunk 2 above)
Comment (Tazmania40) [reply src/surface/gui/gui/widgets/flood_visual.py:39]: Added an `else` clause with a warning that it reset itself.

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]
Rename `flood_visual.py` to `flood_warning.py` to match new class name

**Review by benjaminwp18 — APPROVED** [review]

**Review by InvincibleRMC — APPROVED** [review]

**Pull request merged**